src/box/txn_limbo.c

```diff
@@ -72,6 +72,21 @@ txn_limbo_is_ro(struct txn_limbo *limbo)
 	       (limbo->owner_id != instance_id || txn_limbo_is_frozen(limbo));
 }
 
+void
+txn_limbo_complete(struct txn *txn, bool is_success)
+{
+	/*
+	 * Some rollback/commit triggers require the in_txn fiber
+	 * variable to be set.
+	 */
+	fiber_set_txn(fiber(), txn);
```

Comment:
The code seem to be missing `assert(in_txn() == NULL)`.

Reply:
Agree. It indeed looks like, that `in_txn` cannot be set before confirming everything in limbo. It's removed before writing to journal (because we cannot yield while `in_txn()` is set?) and is returned only in case we need to run triggers.

Reply:
Fixed